File: JS Fundamentals/Basic Syntax, Conditional Statements and Loops - Exercise/vacation.js
function vacation(group, type, day) {
    let price = 0;
    let total_price = 0;
    if (type === `Students`) {
        if (day === 'Friday'){
            price = 8.45
        } else if (day === 'Saturday'){
            price = 9.80
        } else if (day === 'Sunday'){
            price = 10.46
        }
        if (group >= 30){
            total_price = (group * price) * 0.85
        } else {
            total_price = (group * price)
        }
    } else if (type === 'Business'){
        if (day === 'Friday'){
            price = 10.90
        } else if (day === 'Saturday'){
            price = 15.60
        } else if (day === 'Sunday'){
            price = 16
        }
        if (group >= 100){
            total_price = (group * price) - (10 * price)
        } else {
            total_price = group * price
        }
    } else if (type === 'Regular'){
        if (day === 'Friday'){
            price = 15
        } else if (day === 'Saturday'){
            price = 20
        } else if (day === 'Sunday'){
            price = 22.50
        }
        if (group >= 10 && group <= 20){
            total_price = (group * price) * 0.95
        } else {
            total_price = (group * price)
        }
    }
    console.log(`Total price: ${total_price.toFixed(2)}`)
}

vacation(30, "Students", "Sunday");
vacation(40, "Regular", "Saturday");

File: JavaFundamentals/src/BasicSyntaxConditionalStatementsAndLoops/Exercise/Vacation.java
package BasicSyntaxConditionalStatementsAndLoops.Exercise;

import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int groupOfPeople = Integer.parseInt(scanner.nextLine());
        String groupType = scanner.nextLine();
        String day = scanner.nextLine();

        double price = 0;
        switch (day) {
            case "Friday" -> {
                switch (groupType) {
                    case "Students" -> price = 8.45;
                    case "Business" -> price = 10.90;
                    case "Regular" -> price = 15;
                }
            }
            case "Saturday" -> {
                switch (groupType) {
                    case "Students" -> price = 9.80;
                    case "Business" -> price = 15.60;
                    case "Regular" -> price = 20;
                }
            }
            case "Sunday" -> {
                switch (groupType) {
                    case "Students" -> price = 10.46;
                    case "Business" -> price = 16;
                    case "Regular" -> price = 22.50;
                }
            }
        }

        double totalPrice = groupOfPeople * price;

        if (groupType.equals("Students") && groupOfPeople >= 30) {
            totalPrice = totalPrice * 0.85;
        }

        if (groupType.equals("Business") && groupOfPeople >= 100) {
            totalPrice = totalPrice - (10 * price);
        }

        if (groupType.equals("Regular") && groupOfPeople >= 10 && groupOfPeople <= 20) {
            totalPrice = totalPrice * 0.95;
        }

        System.out.printf("Total price: %.2f", totalPrice);
    }
}